### PR TITLE
[AD-195] Implement CLI for the metadata persistence - Part 2

### PR DESCRIPTION
**Pull request by birschick-bq**
### Summary

[AD-195](https://bitquill.atlassian.net/browse/AD-195) Implement CLI for the metadata persistence - Part 2

### Description

Add support for:

1. `-l` "list" - lists persisted schema
1. `-b` "list tables" - lists persisted table schemas
1. `-e` "export" - exports one or more table schema to JSON.
1. `-i` "import" - import table schema and update persisted schema.
1. `-o` "output-file" - outputs exported table schema to a file, instead of standard out.
1. Updated documentation.

### Related Issue

https://bitquill.atlassian.net/browse/AD-195

### Additional Reviewers
@birschick-bq
@mitchell-elholm
@alexey-temnikov 
@andiem-bq

### CODE REVIEW

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbMain.java [hunk 1]
```diff
@@ -1,560 +1,1011 @@
...
+                    + " the schema is written to stdout. Use the --output option to write to a file."
+                    + " The output format is JSON.";
+    private static final String IMPORT_OPTION_DESCRIPTION =
+            "Imports the schema from <file-name> in your home directory. The schema will be imported using the"
+                    + " <schema-name> and a new version will be added - replacing the existing"
+                    + " schema. The expected input format is JSON.";
+    private static final String OUTPUT_OPTION_DESCRIPTION =
+            "Write the exported schema to <file-name> in your home directory (instead of stdout)."
+                    + " This will overwrite any existing file with the same name";
+    // Messages string contants
```
Comment (andiemontoyeah) [inline src/main/java/software/amazon/documentdb/jdbc/DocumentDbMain.java]: NIT: typo in comment - contants -> constants

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbMain.java [hunk 2]
```diff
@@ -1,560 +1,1011 @@
...
+        } catch (Exception e) {
+            LOGGER.error(
+                    "Unexpected exception: '{}'",
+                    e.getMessage(),
+                    e);
+        }
+    }
+
+    static void handleCommandLine(final String[] args, final StringBuilder output)
+            throws ParseException, SQLException {
```
Comment (andiemontoyeah) [inline src/main/java/software/amazon/documentdb/jdbc/DocumentDbMain.java]: can SqlException be thrown here?

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbMainTest.java [hunk 3]
```diff
@@ -291,9 +590,207 @@ void testVersionOption() throws SQLException, ParseException {
...
+            Assertions.assertEquals("Output file name must not be a directory.", output.toString());
+        } finally {
+            Assertions.assertTrue(directoryPath.toFile().delete());
+        }
+    }
+
+    @ParameterizedTest(name = "testExportFileToDirectoryError - [{index}] - {arguments}")
+    @MethodSource("getTestEnvironments")
+    void testImportFileNotExistsError(final DocumentDbTestEnvironment testEnvironment)
+            throws SQLException, ParseException, IOException {
```
Comment (andiemontoyeah) [inline src/test/java/software/amazon/documentdb/jdbc/DocumentDbMainTest.java]: IOException wont get thrown here

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbMainTest.java [hunk 4]
```diff
@@ -58,14 +81,22 @@ static void beforeAll() throws Exception {
...
+            writer.remove(CUSTOM_SCHEMA);
+        }
+        properties = null;
+    }
+
     @AfterAll
     static void afterAll() throws Exception {
         for (DocumentDbTestEnvironment environment : getTestEnvironments()
                 .collect(Collectors.toList())) {
             final DocumentDbConnectionProperties properties = DocumentDbConnectionProperties
```
Comment (andiemontoyeah) [inline src/test/java/software/amazon/documentdb/jdbc/DocumentDbMainTest.java]: properties does not get used anymore

**Comment by alexey-temnikov** [comment]
1. Options s,d,u do not have description.
2. Each option in "Missing required options:" should be starting with the new line for the readability.

File: src/markdown/schema/manage-schema-cli.md [hunk 5]
```diff
@@ -17,6 +17,9 @@ options must be provided.
 |Option|Description|
 |---:|---|
 | <span style="white-space: nowrap;">`-g`, <br>`--generate-new`</span> | Generates a new schema for the database. This will have the effect of replacing an existing schema of the same name, if it exists. |
+| <span style="white-space: nowrap;">`-e`, <br>`--export <[table-name[,...]]>`</span> | Exports the schema to for SQL tables named `[<table-name>[,<table-name>[…]]]`. If no `<table-name>` are given, all table schema will be exported. By default, the schema is written to `stdout`. Use the `-o` option to write to a file. The output format is JSON. |
+| <span style="white-space: nowrap;">`-i`, <br>`--import <file-name>`</span> | Imports the schema from `<file-name>` in your home directory. The schema will be imported using the `<schema-name>` and a new version will be added - replacing the existing schema. The expected input format is JSON. |
+| <span style="white-space: nowrap;">`-l`, <br>`--list`</span> | Lists the schema names, version and table names available in the schema repository.\n"
```
Comment (alexey-temnikov) [inline src/markdown/schema/manage-schema-cli.md]: The output format (comma separated) is not intuitive. I was expecting only the list of the schemas.
Suggesting to include example, and add details that schema is represented on each line (not comma separated)

**Comment by alexey-temnikov** [comment]
Should we add --version support when we are removing schemas?

File: src/markdown/schema/manage-schema-cli.md (hunk 5 above)
Comment (birschick-bq) [reply src/markdown/schema/manage-schema-cli.md]: @alexey-temnikov So, we do want to show the list of associated table names.

Suggested format any better?

```
Name: _default, Version: 1, Modified: 2021-06-11T12:34:56-08:00
- table1
- table1_array
- table1_subdoc
- table2
- table2_etc
```

**Review by andiemontoyeah — APPROVED** [review]
Nothing else from me! 😄

File: src/markdown/schema/manage-schema-cli.md (hunk 5 above)
Comment (birschick-bq) [reply src/markdown/schema/manage-schema-cli.md]: Separate the `--list-schema` and `--list-tables`` commands.

`--list-schema` has this output format:

```
Name: _default, Version: 1, Modified: 2021-06-11T12:34:56-08:00
```

`-list-tables` has this output format

```
products
products_additional_tarriffs
products_for
products_limits
products_limits_data
products_limits_sms
products_limits_voice
products_type
projects
```

**Pull request merged**